packages/stryker/src/MutantTestMatcher.ts

```diff
@@ -1,19 +1,35 @@
 import { getLogger } from 'log4js';
 import * as _ from 'lodash';
-import { RunResult, CoverageCollection, StatementMap, CoveragePerTestResult } from 'stryker-api/test_runner';
+import { RunResult, CoverageCollection, StatementMap, CoveragePerTestResult, CoverageResult } from 'stryker-api/test_runner';
 import { Location, StrykerOptions, File, TextFile } from 'stryker-api/core';
 import { MatchedMutant } from 'stryker-api/report';
 import { Mutant } from 'stryker-api/mutant';
 import TestableMutant from './TestableMutant';
 import StrictReporter from './reporters/StrictReporter';
-import { StatementMapDictionary } from './transpiler/CoverageInstrumenterTranspiler';
+import { FileCoverageDataDictionary, FileCoverageMaps } from './transpiler/CoverageInstrumenterTranspiler';
 import { filterEmpty } from './utils/objectUtils';
 import SourceFile from './SourceFile';
+import { Range } from 'istanbul-lib-coverage';
+
+
+enum CoveredCodeIndicatorType {
+  function,
+  statement
+}
+
+/**
+ * Represents a location inside the coverage data of a file
+ * Either the function map, or statement map
+ */
+interface CoveredCodeIndicator {
+  type: CoveredCodeIndicatorType;
+  index: string;
```

Comment:
Why set the index here and not the actual value? Right now it looks like we set the index to either the `statementIndex` or `functionIndex` and then read it later on. That would also remove the need for the vague name 'index'

Reply:
What actual value? We need thr index, because the index in the statement/fn map is also the key of the coverage data objects (`s` and `f`)

Reply:
We set the index in `findMatchingCoveringIndicator` and in `isCoveredByCoverageCollection` we read the value with the index that we have set. Why can't we read the value in `findMatchingCoveringIndicator` and add that value to the object instead of the index?